catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -639,17 +654,27 @@ private String transformToXml(String xhtml) {
             LOGGER.debug(e.getMessage(), e);
         }
         if (xmlReader != null) {
-            try {
-                Writer xml = new StringWriter();
+            try (TemporaryFileBackedOutputStream xmlOutStream = new TemporaryFileBackedOutputStream();
+                    InputStream xhtmlInStream = xhtml.asByteSource()
+                            .openStream()) {
                 Transformer transformer = templates.newTransformer();
-                transformer.transform(new SAXSource(xmlReader,
-                        new InputSource(new StringReader(xhtml))), new StreamResult(xml));
-                return xml.toString();
-            } catch (TransformerException e) {
+                transformer.transform(new SAXSource(xmlReader, new InputSource(xhtmlInStream)),
+                        new StreamResult(xmlOutStream));
+                //we should not be doing this and should be returning the stream instead
```

Comment:
Do you want to make the change now to return a stream?  In the short run it probably doesn't matter because the MetacardCreator string to populate the Metacard's metadata attribute.